python/ray/ray_constants.py

```diff
@@ -49,7 +53,47 @@ def env_integer(key, default):
 # The maximum resource quantity that is allowed. TODO(rkn): This could be
 # relaxed, but the current implementation of the node manager will be slower
 # for large resource quantities due to bookkeeping of specific resource IDs.
-MAX_RESOURCE_QUANTITY = 512
+MAX_RESOURCE_QUANTITY = 10000
+
+# Each memory "resource" counts as this many bytes of memory.
+MEMORY_RESOURCE_UNIT_BYTES = 100 * 1024 * 1024
```

Comment:
I think it'd be clearer to use powers of 10 and to have the minimum object store memory by `10**8`.

`ray.init()` reports stuff in GB, but currently `@ray.remote` computes units in powers of 2.

Reply:
It's a bit of an internal detail except that as a user it is exposed through error messages like

*2019-08-09 23:09:33,940 ERROR worker.py:1781 -- The task with ID ffffffffffff7d58f41501000000 is infeasible and cannot currently be executed. It requires {object_store_memory,96.000000} for execution and {object_store_memory,96.000000},{CPU,1.000000} for placement. Check the client table to view node resources.*

I was expecting the number **96** in that error message to be **100**. The workload that produced this is

```python
import ray

ray.init()

@ray.remote(object_store_memory=10**10) 
class Foo: 
    def __init__(self): 
        pass

f = Foo.remote()
```